bdns/dns_test.go

```diff
@@ -268,8 +272,10 @@ func TestDNSOneServer(t *testing.T) {
 
 	obj := NewTest(time.Second*10, staticProvider, metrics.NoopRegisterer, clock.NewFake(), 1, blog.UseMock(), nil)
 
-	_, err = obj.LookupHost(context.Background(), "cps.letsencrypt.org")
-
+	_, resolvers, err := obj.LookupHost(context.Background(), "cps.letsencrypt.org")
+	slices.Sort(resolvers)
```

Comment:
This is a very tiny nit, but: I'd put the sort _immediately_ before the AssertDeepEquals, so that it's clear to a reader why we're sorting. This is less important here where they're only one line apart, but much more important in places like line 337 where there's a whole paragraph between the sort and the reason we're sorting.